bin/build.mjs

```diff
@@ -89,32 +96,53 @@ async function buildPackage(absWorkingDir) {
     const esmContext = await esbuild.context({
       ...buildOptions,
       ...entryPoints('esm'),
-      outdir: 'lib',
       absWorkingDir,
     });
 
-    await esmContext.watch();
+    const extraContext =
+      packageJson.name === '@pixiv/vrm-viewer'
+        ? (await esbuild.context(vrmViewOptions(buildOptions))).watch()
+        : Promise.resolve();
+
+    await Promise.all([esmContext.watch(), extraContext]);
     await esmContext.serve({ servedir: '.', port: PORT });
   }
 
   async function build() {
     await esbuild.build({
       ...buildOptions,
       ...entryPoints('esm'),
-      outdir: 'lib',
     });
 
     await esbuild.build({
       ...buildOptions,
       ...entryPoints('cjs'),
-      outdir: 'lib',
       outExtension: {
         // eslint-disable-next-line @typescript-eslint/naming-convention
         '.js': '.cjs',
       },
     });
+
+    if (packageJson.name === '@pixiv/vrm-viewer') {
+      await esbuild.build(vrmViewOptions(buildOptions));
+    }
   }
 
   // == main =========================================================================================
   await (SERVE ? serve : build)();
 }
+
+/**
+ * @param {import('esbuild').BuildOptions} baseBuildOptions
+ * @returns {import('esbuild').BuildOptions}
+ */
+function vrmViewOptions(baseBuildOptions) {
+  return {
+    ...baseBuildOptions,
+    format: 'esm',
+    entryPoints: {
+      [`vrm-viewer${minifySuffix}`]: 'src/index.ts',
+    },
+    external: [], // include three
+  };
+}
```

Comment:
Please clarify the intention why we have to separate options only for vrm-viewer.

I guess it's because:

- we don't have to output `cjs` builds
- we want to specify that the output filename is `vrm-viewer.js` or `vrm-viewer.min.js`
- we want to include Three.js in the build

Reply:
> we don't have to output cjs builds
YES, but can be avoid by reusing the esm context.
> we want to specify that the output filename is vrm-viewer.js or vrm-viewer.min.js
YES, but can be avoid by reusing existing context.
> we want to include Three.js in the build
YES, this is the main reason since `external` is fixed per context.

Reply:
would you put them in the code as comments?

Reply:
https://github.com/pixiv/three-vrm/pull/1395/commits/76a31409fb74b22791cc96cf7b30a3e454d87607